### ISSUE TITLE
SSL 1.1 issues on Debian Stretch

### ISSUE THREAD

**Issue by rburkholder** [bisdn/rofl-common#83]
I have these ssl libraries installed on my system:
```
# ls -al /usr/lib/x86_64-linux-gnu/ | grep ssl
-rw-r--r-- 1 root root   737466 Feb 16 13:57 libssl.a
lrwxrwxrwx 1 root root       13 Feb 16 13:57 libssl.so -> libssl.so.1.1
-rw-r--r-- 1 root root   431232 Jan 26 11:39 libssl.so.1.0.2
-rw-r--r-- 1 root root   442920 Feb 16 13:57 libssl.so.1.1
drwxr-xr-x 1 root root       14 Dec  7 14:20 openssl-1.0.2
# ls -al /usr/lib/x86_64-linux-gnu/ | grep crypto
-rw-r--r-- 1 root root  4799548 Feb 16 13:57 libcrypto.a
lrwxrwxrwx 1 root root       16 Feb 16 13:57 libcrypto.so -> libcrypto.so.1.1
-rw-r--r-- 1 root root  2492192 Jan 26 11:39 libcrypto.so.1.0.2
-rw-r--r-- 1 root root  2686448 Feb 16 13:57 libcrypto.so.1.1
lrwxrwxrwx 1 root root       18 Dec  4 16:37 libk5crypto.so.3 -> libk5crypto.so.3.1
-rw-r--r-- 1 root root   203656 Dec  4 16:37 libk5crypto.so.3.1
```

I need to run the following to ensure ssl libraries are included, I see the -lcrypto is applied properly (configure seems to miss somehow):

```
LIBS=-lssl ../configure
```

In addition, with SSL 1.1, I run the following before I do the autogen.sh:

```
sed -i 's/TLSv1_2_method()/TLS_method()/' src/rofl/common/crofsock.c
```

Some relevant information:

https://www.openssl.org/docs/man1.1.0/ssl/SSL_CTX_new.html

**Comment by toanju** [comment]
Agree this is pretty broken right now. As soon as we find some time we will fix it. PRs are welcome though :)

**Comment by toanju** [comment]
We just merged #95 to build with OpenSSL 1.1. We did not yet test the SSL connectivity though, but at least the library builds again and the unencrypted parts work fine.
